e2e/malicious-join/malicious-join.go

```diff
@@ -50,48 +58,44 @@ func main() {
 		TestCases: make(map[string]testCaseOutput),
 	}
 	for name, tc := range testCases {
-		fmt.Printf("Running testcase %s\n", name)
+		log.With(zap.String("testcase", name)).Infof("Running testcase")
 
-		err := tc.fn(*attVariant, *csp, *jsEndpoint)
+		err := tc.fn(*attVariant, *csp, *jsEndpoint, log)
 
 		switch {
 		case err == nil && tc.wantErr:
-			fmt.Printf("Test case %s failed: Expected error but got none\n", name)
+			log.With(zap.Error(err), zap.String("testcase", name)).Errorf("Test case failed: Expected error but got none")
 			testOutput.TestCases[name] = testCaseOutput{
 				Passed:  false,
 				Message: "Expected error but got none",
 			}
 			allPassed = false
 		case !tc.wantErr && err != nil:
-			fmt.Printf("Test case %s failed: Got unexpected error: %s\n", name, err)
+			log.With(zap.Error(err), zap.String("testcase", name)).Errorf("Test case failed: Got unexpected error")
 			testOutput.TestCases[name] = testCaseOutput{
 				Passed:  false,
 				Message: fmt.Sprintf("Got unexpected error: %s", err),
 			}
 			allPassed = false
 		case tc.wantErr && err != nil:
-			fmt.Printf("Test case %s succeeded\n", name)
+			log.With(zap.String("testcase", name)).Infof("Test case succeeded")
 			testOutput.TestCases[name] = testCaseOutput{
 				Passed:  true,
 				Message: fmt.Sprintf("Got expected error: %s", err),
 			}
 		case !tc.wantErr && err == nil:
-			fmt.Printf("Test case %s succeeded\n", name)
+			log.With(zap.String("testcase", name)).Infof("Test case succeeded")
 			testOutput.TestCases[name] = testCaseOutput{
 				Passed:  true,
 				Message: "No error, as expected",
 			}
 		default:
-			panic("invalid result")
+			log.With(zap.String("testcase", name)).Fatalf("invalid result")
 		}
 	}
 
 	testOutput.AllPassed = allPassed
-	out, err := json.Marshal(testOutput)
-	if err != nil {
-		panic(fmt.Sprintf("marshalling test output: %s", err))
-	}
-	fmt.Println(string(out))
+	log.With(zap.Any("result", testOutput)).Infof("Test completed")
```

Comment:
I assume, each zap logger instance has it's own buffer? Otherwise, we might want to flush before here to ensure that it's the last log.

Reply:
See https://github.com/edgelesssys/constellation/blob/7cdb7fd4555921eab1c149a0b4321c33067e6916/e2e/malicious-join/malicious-join.go#L29